soc/arm/microchip_mec/common/spigen/mec_spi_gen.py

```diff
@@ -0,0 +1,510 @@
+    parser.add_argument("-f",
+                        type=int,
+                        dest="spi_freq_mhz",
+                        default=12,
+                        help="SPI frequency: 12, 16, 24, or 48 MHz")
+    parser.add_argument("-r",
+                        type=str,
+                        dest="spi_read_mode",
+                        default="fast",
+                        help="SPI read mode: normal, fast, dual or quad")
+    parser.add_argument("-m",
+                        type=int,
+                        dest="spi_mode",
+                        default=0,
+                        help="SPI signalling mode 3-bit field: 0-7")
+    parser.add_argument("--drvstr",
+                        type=str,
+                        dest="spi_drive_strength",
+                        default="1x",
+                        help="SPI pin driver strength multiplier encoded")
+    parser.add_argument("--slewrate",
+                        type=str,
+                        dest="spi_slew_rate",
+                        default="slow",
+                        help="SPI pins slew rate")
+    parser.add_argument("--fill",
+                        dest="fill",
+                        action='store_true',
+                        help="Fill with 0xFF to flash size")
+    parser.add_argument("-v",
+                        dest="verbose",
+                        action='store_true',
+                        help="Enable messages to console")
+
+    ret_args = parser.parse_args()
+
+    return ret_args
+
+def main():
+    """MEC SPI Gen"""
+
+    args = parse_args()
+
+    verbose_mode = args.verbose
+
+    if verbose_mode:
+        print("Command line arguments/defaults")
+        print("  chip = {0}".format(args.chip))
+        print("  infilename = {0}".format(args.infilename))
+        print("  outfilename = {0}".format(args.outfilename))
+        print("  SPI size (kilobytes) = {0}".format(args.spi_size_kb))
+        print("  Entry point address = {0}".format(args.entry_point))
+        print("  SPI frequency MHz = {0}".format(args.spi_freq_mhz))
+        print("  SPI Read Mode = {0}".format(args.spi_read_mode))
+        print("  SPI Signalling Mode = {0}".format(args.spi_mode))
+        print("  SPI drive strength = {0}".format(args.spi_drive_strength))
+        print("  SPI slew rate fast = {0}".format(args.spi_slew_rate))
+        print("  Verbose = {0}".format(args.verbose))
+
+    if args.infilename is None:
+        if verbose_mode:
+            print("ERROR: Specify input binary file name with -i")
+        sys.exit(-1)
+
+    if args.outfilename is None:
+        if verbose_mode:
+            print("ERROR: Specify output binary file name with -o")
+        sys.exit(-1)
+
+    chip = CHIP_DEFAULT
+    if args.chip in ["mec152x", "mec172x"]:
+        chip = args.chip
+    else:
+        if verbose_mode:
+            print("ERROR: Unknown chip {0}. Use default chip: {1}".format(args.chip, CHIP_DEFAULT))
```

Comment:
This pattern (discard an explicit flag, fall back to default, only warn on verbose=True) seems like asking for trouble. If the flag value is not valid make the script fail explicitly and return an error.